ERCS/erc-xxxx.md

```diff
@@ -0,0 +1,243 @@
+such as `acceptAccount`, `acceptPaymaster`, `sigFailAccount` and `sigFailPaymaster`, is considered a success.
+
+If it fails to find such a value, returns an error message with the detailed description of the failure reason.
+
+Parameters:
+
+1. OBJECT - The RIP-7560 transaction object.
+   The `validationGasLimit`, paymasterValidationGasLimit, `paymasterGasLimit` and `callGasLimit` fields are optional.
+2. QUANTITY | TAG - integer block number, or the string "latest", "earliest", "pending", "safe" or "finalized"
+3. OBJECT - State override set
+   The State Override Set option allows you to change the state of a contract before executing the call. This means you
+   can modify the values of variables stored in the contract, such as balances and approvals for that call without
+   actually modifying the contract on the blockchain.
+   This behavior is equivalent to the one defined for the `eth_call` RPC method.
+
+Returns:
+
+1. Object
+
+| Name                        | Type     |
+|-----------------------------|----------|
+| validationGasLimit          | QUANTITY |
+| paymasterValidationGasLimit | QUANTITY |
+| paymasterPostOpGasLimit     | QUANTITY |
+| callGasLimit                | QUANTITY |
+
+Error:
+
+DATA - The revert data of the first reverted frame.
+CODE - The error code indicating the type of error, which may include the entity that caused the revert on-chain.
+MESSAGE - The human-readable error that may include a decoding of the `DATA` field if possible.
+
+#### Notes on implementation details
+
+As mentioned in the RIP-7560, the `sender` and `paymaster` contracts should not revert on the validation failure
+and should make calls to `sigFailAccount` and `sigFailPaymaster` respectively
+in order to support `eth_estimateRip7560TransactionGas`.
+
+The recommended way to achieve this behavior for Smart Contract Accounts and Paymasters is to compare the `signature`
+parameter to a predetermined "dummy signature" and to call a `sigFail` callback in case the values match.
+
+### Create a new JSON-RPC API method `eth_traceRip7560Validation`
+
+Only executes the validation phase of the RIP-7560 transaction and returns the tracing results of this execution.
+This is done in order to allow other clients to determine
+whether all contracts used within the validation phase of this transaction are compliant with the rules
+defined in the [ERC-7562](https://eips.ethereum.org/EIPS/eip-7562).
+
+Parameters:
+
+1. OBJECT - The RIP-7560 transaction object.
+2. QUANTITY | TAG - a block number, or the string "earliest", "latest", "pending", "safe" or "finalized", as in the
+   default block parameter.
+
+Returns:
+
+1. OBJECT - the tracing result of executing the entire validation phase of the RIP-7560 transaction.\
+The exact shape of the returned object is not strictly defined by this standard and can be decided
+by different implementation.
+
+### Error Codes
```

Comment:
If we do add this tracing RPC, doesn't it need to also return proper error codes when ERC-7562 rules are violated (so the transaction isn't even fully validated because it won't be accepted due to the 7562 violation)?

Reply:
Currently, we have a tracer implemented in Go but the parser is still in TypeScript inside bundler, and this API is used by a bundler who gets the stack trace, parses it and finds ERC-7562 violations. So unless we implement parsing in Go we won't be able to return violation errors from the node yet.